api/controllers/install/infra.go

```diff
@@ -2,18 +2,34 @@ package install
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"runtime/debug"
 
 	"github.com/replicatedhq/embedded-cluster/api/types"
 )
 
-func (c *InstallController) SetupInfra(ctx context.Context) (finalErr error) {
-	if c.stateMachine.CurrentState() == StatePreflightsFailed {
-		err := c.bypassPreflights(ctx)
+var (
+	ErrPreflightChecksFailed      = errors.New("preflight checks failed")
+	ErrPreflightChecksNotComplete = errors.New("preflight checks not complete")
+)
+
+func (c *InstallController) SetupInfra(ctx context.Context, ignorePreflightFailures bool) (finalErr error) {
+	currentState := c.stateMachine.CurrentState()
+
+	// Check if preflights are complete
+	switch currentState {
```

Comment:
why is this switch statement necessary? it seems like something we should leave to the state machine to return conflict errors based on state, and should simply pass the `ignorePreflightFailures` bool to the `c.bypassPreflights` function.